### PR TITLE
Fix Daylighting EMS lag Part1 - move surface EMS inits from InitSurfaceHeatBalance to InitHeatBalance

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
When using EMS actuators to change window construction type, the impact on daylighting illuminance was lagged by two timesteps (and possibly not fully accounted for). This partially fixes that by calling the surface-level EMS inits earlier in the heat balance. Partially addresses #4266, maybe #3654, and possibly others - need to search on daylighting issues. The reported construction index and changes in daylighting and solar gains are now in synch.

However, construction changes still do not take effect until one timestep after the EMS program takes action, because the BeginTimestepBeforePredictor calling point happens too late. The final solution is to add a new calling point, but that can't happen now until the next release due to the required IDD change. The new calling point has been added here as a comment line. Suggestion added to new feature request [#6626](https://github.com/NREL/EnergyPlus/issues/6626#issuecomment-421657356).

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes N/A
 - [ ] Documentation changes in place

### CODE REVIEW

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-3** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by nrel-bot-2** [comment]
@mjwitte @lgentile it has been 14 days since this pull request was last updated.

**Comment by mjwitte** [comment]
Moving the existing BeginTimestepBeforePredictor calling point was a bad idea (not really a surprise). Will revert that, and add a unit test to this. Post v9.0 release, a new calling point (or more) can be added, and perhaps some other EMS requests like additional actuators can be addressed at the same time.

File: src/EnergyPlus/HeatBalanceManager.cc [hunk 1]
```diff
@@ -347,6 +347,8 @@ namespace HeatBalanceManager {
             ManageHeatBalanceGetInputFlag = false;
         }
 
+        // Suggest new calling point here ManageEMS(emsCallFromBeginZoneTimestepBeforeInitHeatBalance, anyRan);
```
Comment (Myoldmopar) [inline src/EnergyPlus/HeatBalanceManager.cc:350]: Will be nice.

File: src/EnergyPlus/DaylightingManager.cc [hunk 2]
```diff
@@ -234,6 +246,25 @@ namespace DaylightingManager {
     {
         // this will need a lot more, but it is a start
         ZoneDaylight.deallocate();
+        CalcDayltghCoefficients_firstTime = true;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DaylightingManager.cc:249]: Excellent additions here to the clear_state

File: src/EnergyPlus/DaylightingManager.cc [hunk 3]
```diff
@@ -548,11 +577,11 @@ namespace DaylightingManager {
             "('! <Sky Daylight Factors>, MonthAndDay, Zone Name, Window Name, Daylight Fac: Ref Pt #1, Daylight Fac: Ref Pt #2')");
 
         // FLOW:
-        if (firstTime) {
+        if (CalcDayltghCoefficients_firstTime) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/DaylightingManager.cc:580]: Breaking these into different flags is good.

File: src/EnergyPlus/HeatBalanceManager.cc [hunk 4]
```diff
@@ -5250,6 +5252,11 @@ namespace HeatBalanceManager {
             }
         }
 
+        if (DataGlobals::AnyEnergyManagementSystemInModel) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/HeatBalanceManager.cc:5255]: :+1:

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
I'm OK with the changes here.  I'm building locally and I'll try to run the defect files, but it's a bit difficult to say which issues this will close.

**Comment by mjwitte** [comment]
It doesn't really close any of them, just makes #4266 a little better than it was.

**Comment by Myoldmopar** [comment]
EMS controlled defect file for #4266 is struggling to run with an EMS actuator issue.  I'm just going to move forward with this PR going in.  Thanks @mjwitte

**Comment by mjwitte** [comment]
Sorry, I never shared my other test files - do you want them?

**Pull request merged**